src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -1101,6 +1111,104 @@ SequenceDDLCommandsForTable(Oid relationId)
 }
 
 
+/*
+ * SequenceDependencyCommandList generates commands to record the dependency
+ * of sequences on tables on the worker. This dependency does not exist by
+ * default since the sequences and table are created separately, but it is
+ * necessary to ensure that the sequence is dropped when the table is
+ * dropped.
+ */
+static List *
+SequenceDependencyCommandList(Oid relationId)
+{
+	List *sequenceCommandList = NIL;
+	char *relationName = generate_qualified_relation_name(relationId);
+	HeapTuple tup = NULL;
+	ScanKeyData key[2];
+
+	Relation pgDepend = table_open(DependRelationId, AccessShareLock);
+
+	ScanKeyInit(&key[0],
+				Anum_pg_depend_refclassid,
+				BTEqualStrategyNumber, F_OIDEQ,
+				ObjectIdGetDatum(RelationRelationId));
+	ScanKeyInit(&key[1],
+				Anum_pg_depend_refobjid,
+				BTEqualStrategyNumber, F_OIDEQ,
+				ObjectIdGetDatum(relationId));
+
+	SysScanDesc scan = systable_beginscan(pgDepend, DependReferenceIndexId, true,
+										  NULL, 2, key);
+
+	while (HeapTupleIsValid(tup = systable_getnext(scan)))
+	{
+		Form_pg_depend dependencyRecord = (Form_pg_depend) GETSTRUCT(tup);
+
+		/*
+		 * We assume any auto or internal dependency of a sequence on a column
+		 * must be what we are looking for.  (We need the relkind test because
+		 * indexes can also have auto dependencies on columns.)
+		 */
+		if (dependencyRecord->classid == RelationRelationId &&
```

Comment:
I have some questions/thoughts here:

1-> Why are we interested in DEPENDENCY_INTERNAL case? Does that cover the internal sequence creation for bigserial column?
2-> We could simplify this if check a bit with the usage of some variables maybe. For example it probably makes sense to combine `dependencyRecord->deptype == DEPENDENCY_AUTO` and `get_rel_relkind(dependencyRecord->objid) == RELKIND_SEQUENCE`. I have looked at the specific fields to understand what this if check is doing but for future I would suggest having something like this:

```c
bool isRefObjAColReference = dependencyRecord->refobjsubid != 0;
bool isSeqDepType = (dependencyRecord->deptype == DEPENDENCY_AUTO && get_rel_relkind(dependencyRecord->objid) == RELKIND_SEQUENCE)
 || dependencyRecord->deptype == DEPENDENCY_INTERNAL;
```

Reply:
This is copied from getOwnedSequences_internal, I'll add a note.

Reply:
>This is copied from getOwnedSequences_internal, I'll add a note.

related: https://github.com/citusdata/citus/pull/4203#discussion_r498256254